spring-xd-module/src/main/java/org/springframework/xd/module/core/ModuleFactory.java

```diff
@@ -94,6 +99,10 @@ public Module createModule(ModuleDescriptor moduleDescriptor, ModuleDeploymentPr
 	 */
 	private Module createAndConfigureModuleInstance(ModuleDescriptor moduleDescriptor, ModuleOptions moduleOptions,
 			ModuleDeploymentProperties deploymentProperties) {
+		String name = (String) moduleOptions.asPropertySource().getProperty(MODULE_EXECUTION_FRAMEWORK);
+		if (Processor.MODULE_EXECUTION_FRAMEWORK.equals(name)) {
+			return createSparkModule(moduleDescriptor, moduleOptions, deploymentProperties);
```

Comment:
At this point in the code, there should only be 2 "main" cases here: Simple vs Composed.